### PR TITLE
Improved prune and registry image delete error handling

### LINKED ISSUE #1323: Error rates: Some commands (like prune) don't give any indication that they are running and can take some time
Repro steps:
1. Just open the extension 
2. Try to prune images

Action: vscode-docker.images.prune
Error type: 409
Error Message: (HTTP code 409) unexpected - a prune operation is already running 

Version: 0.8.1
OS: darwin
Product: Visual Studio Code
Product Version: 1.38.1
Language: en

<details>
<summary>Call Stack</summary>

```
extension.bundle.js:157:703034
extension.bundle.js:157:702997
module.exports.m.buildPayload extension.bundle.js:157:703007
IncomingMessage.<anonymous> extension.bundle.js:157:702507
IncomingMessage.emit events.js:187:15
endReadableNT _stream_readable.js:1092:12
process._tickCallback next_tick.js:63:19
```

</details>

### LINKED ISSUE #1683: Error rates: 405 not supported when deleting from registry
We get a lot of bug reports on deleting images from registry failing when it's not allowed. This is not a bug; registries must be explicitly configured to allow deletes.

Examples: https://github.com/microsoft/vscode-docker/issues?q=vscode-docker.registries.deleteImage+unsupported

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes #1323, fixes #1683

### CODE REVIEW

File: src/tree/registries/dockerV2/DockerV2TagTreeItem.ts [hunk 1]
```diff
@@ -28,9 +30,22 @@ export class DockerV2TagTreeItem extends RemoteTagTreeItem {
...
+    public async deleteTreeItemImpl(context: IActionContext): Promise<void> {
         const digest = await this.getDigest();
         const url = `v2/${this.parent.repoName}/manifests/${digest}`;
-        await registryRequest(this.parent, 'DELETE', url);
+
+        try {
+            await registryRequest(this.parent, 'DELETE', url);
+        } catch (error) {
+            const errorType: string = parseError(error).errorType.toLowerCase();
+            if (errorType === '405' || errorType === 'method not supported') { // TODO: is this error type right at all?
```
Comment (bwateratmsft) [inline src/tree/registries/dockerV2/DockerV2TagTreeItem.ts]: Do

File: src/commands/containers/pruneContainers.ts [hunk 2]
```diff
@@ -15,13 +15,18 @@ export async function pruneContainers(context: IActionContext): Promise<void> {
...
 
-    const numDeleted = (result.ContainersDeleted || []).length;
-    const mbReclaimed = convertToMB(result.SpaceReclaimed);
-    let message = localize('vscode-docker.commands.containers.prune.removed', 'Removed {0} container(s) and reclaimed {1}MB of space.', numDeleted, mbReclaimed);
-    // don't wait
-    /* eslint-disable-next-line @typescript-eslint/no-floating-promises */
-    window.showInformationMessage(message);
+            const numDeleted = (result.ContainersDeleted || []).length;
+            const mbReclaimed = convertToMB(result.SpaceReclaimed);
+            let message = localize('vscode-docker.commands.containers.prune.removed', 'Removed {0} container(s) and reclaimed {1}MB of space.', numDeleted, mbReclaimed);
```
Comment (karolz-ms) [inline src/commands/containers/pruneContainers.ts]: Add a space before disk space unit (MB)?

File: src/commands/images/pruneImages.ts [hunk 3]
```diff
@@ -15,13 +15,18 @@ export async function pruneImages(context: IActionContext): Promise<void> {
...
 
-    const numDeleted = (result.ImagesDeleted || []).length;
-    const mbReclaimed = convertToMB(result.SpaceReclaimed);
-    let message = localize('vscode-docker.commands.images.prune.removed', 'Removed {0} images(s) and reclaimed {1}MB of space.', numDeleted, mbReclaimed);
-    // don't wait
-    /* eslint-disable-next-line @typescript-eslint/no-floating-promises */
-    window.showInformationMessage(message);
+            const numDeleted = (result.ImagesDeleted || []).length;
+            const mbReclaimed = convertToMB(result.SpaceReclaimed);
+            let message = localize('vscode-docker.commands.images.prune.removed', 'Removed {0} images(s) and reclaimed {1}MB of space.', numDeleted, mbReclaimed);
```
Comment (karolz-ms) [inline src/commands/images/pruneImages.ts]: Add a space before disk space unit (MB)?

File: src/commands/pruneSystem.ts [hunk 4]
```diff
@@ -15,21 +15,26 @@ export async function pruneSystem(context: IActionContext): Promise<void> {
...
-    const volumesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneVolumes(), context);
-    /* eslint-enable @typescript-eslint/promise-function-async */
+    await vscode.window.withProgress(
+        { location: vscode.ProgressLocation.Notification, title: localize('vscode-docker.commands.pruneSystem.pruning', 'Pruning system...') },
+        async () => {
+            /* eslint-disable @typescript-eslint/promise-function-async */
+            const containersResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneContainers(), context);
+            const imagesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneImages(), context);
+            const networksResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneNetworks(), context);
+            const volumesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneVolumes(), context);
```
Comment (karolz-ms) [inline src/commands/pruneSystem.ts]: Not sure if you want to address it with this change, but we got user feedback we should ask for extra confirmation about pruning volumes, as they might contain useful data.

Comment (bwateratmsft) [reply src/commands/pruneSystem.ts]: I think probably the right way is to add an extra warning to the prompts--that pruning volumes can result in lost data.

**Pull request merged**